droopy: use python 3.7 for macpath

sternenseemann opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

The `macpath` module droopy uses got removed in python 3.8. Use Python 2.7 for droopy, since droopy originally targets that version.

cc @Profpatsch 

[Upstream issue](https://github.com/stackp/Droopy/issues/27), main problem is that `droopy` is unmaintained.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Mindavi commented:
Have you tried using the suggested method of replacing `macpath` with `os.path`? That seems like a better solution to me. Are you still using Mac OS X 9 or lower? It seems like the `macpath` module is mainly focused on support for that.

I'm not sure what the stance of nixpkgs / the nixpkgs maintainers is, but I'd like to see python 2.7 usage reduced to a minimum if possible.

I'd wager it wouldn't be a big path to move to `os.path` usage, but I'm not sure about that.

jonringer commented:
I would much rather have python3.7.

python2 has been EOL for 2/3's of a year now

sternenseemann commented:
Good point, then I'd say we go with python 3.7. I'd rather start patching later than sooner personally.

sternenseemann committed:
droopy: use python 3.7 for macpath

droopy uses the macpath module which got removed in python 3.8. Pinning
the python version to 3.7 fixes the ensuing runtime failure (for now).

jonringer reviewed (approved):
diff LGTM

```
https://github.com/NixOS/nixpkgs/pull/96671
1 package built:
droopy
```

Profpatsch commented:
thx sterni
